app/models/user_block.rb

```diff
@@ -28,6 +29,8 @@
 class UserBlock < ApplicationRecord
   validate :moderator_permissions
   validates :reason, :characters => true
+  validates :deactivates_at, :comparison => { :greater_than_or_equal_to => :ends_at }, :if => -> { !needs_view }
```

Comment:
I think this could be `:unless => -> { needs_view }`?

Reply:
updated to `:unless`